wordpress-mobile/WordPress-iOS
# Introduce rich notification support for Comments

### Description
This PR merges changes so that _Comment_ notifications are presented in a richer way in both the _Short Look_ & _Long Look_ (evinced via force-touching the short look). 

The contents of this long-running feature branch (i.e., `feature/rich-notifications`) have absorbed upstream content at least every two weeks. Moreover, each of the constituent commits have already been reviewed & approved.

### To test

The branch should build & existing tests should pass.

As with prior manual tests of push notifications, it should be possible to fabricate a push notification for testing on-device. The payload for such a notification should include: `mutable-content`,  `note_id`, `category` & `type`. Please contact me if you need assistance with that.

### Expected Results

_Short Look_

![a](https://user-images.githubusercontent.com/221062/45118699-108af200-b10e-11e8-99a2-b544f1095379.jpg)

_Long Look (Above fold)_

![b1](https://user-images.githubusercontent.com/221062/45118700-108af200-b10e-11e8-866f-d630bf379a68.png)
